// @google/adk/dist/cjs/models/base_llm.js
/**
  * @license
  * Copyright 2025 Google LLC
  * SPDX-License-Identifier: Apache-2.0
  */

"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var base_llm_exports = {};
__export(base_llm_exports, {
  BaseLlm: () => BaseLlm
});
module.exports = __toCommonJS(base_llm_exports);
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */
class BaseLlm {
  /**
   * Creates an instance of BaseLLM.
   *
   * @param model The name of the LLM, e.g. gemini-1.5-flash or
   *     gemini-1.5-flash-001.
   */
  constructor(model) {
    this.model = model;
  }
  /**
   * Appends a user content, so that model can continue to output.
   *
   * @param llmRequest LlmRequest, the request to send to the LLM.
   */
  maybeAppendUserContent(llmRequest) {
    var _a;
    if (llmRequest.contents.length === 0) {
      llmRequest.contents.push({
        role: "user",
        parts: [
          { text: "Handle the requests as specified in the System Instruction." }
        ]
      });
    }
    if (((_a = llmRequest.contents[llmRequest.contents.length - 1]) == null ? void 0 : _a.role) !== "user") {
      llmRequest.contents.push({
        role: "user",
        parts: [{
          text: "Continue processing previous requests as instructed. Exit or provide a summary if no more outputs are needed."
        }]
      });
    }
  }
}
/**
 * List of supported models in regex for LlmRegistry.
 */
BaseLlm.supportedModels = [];
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BaseLlm
});
