// @google/adk/dist/esm/models/base_llm.js
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */
class BaseLlm {
  /**
   * Creates an instance of BaseLLM.
   *
   * @param model The name of the LLM, e.g. gemini-1.5-flash or
   *     gemini-1.5-flash-001.
   */
  constructor(model) {
    this.model = model;
  }
  /**
   * Appends a user content, so that model can continue to output.
   *
   * @param llmRequest LlmRequest, the request to send to the LLM.
   */
  maybeAppendUserContent(llmRequest) {
    var _a;
    if (llmRequest.contents.length === 0) {
      llmRequest.contents.push({
        role: "user",
        parts: [
          { text: "Handle the requests as specified in the System Instruction." }
        ]
      });
    }
    if (((_a = llmRequest.contents[llmRequest.contents.length - 1]) == null ? void 0 : _a.role) !== "user") {
      llmRequest.contents.push({
        role: "user",
        parts: [{
          text: "Continue processing previous requests as instructed. Exit or provide a summary if no more outputs are needed."
        }]
      });
    }
  }
}
/**
 * List of supported models in regex for LlmRegistry.
 */
BaseLlm.supportedModels = [];
export {
  BaseLlm
};
